README.md

```diff
@@ -1,11 +1,110 @@
 # infra-deployments
 
 ## Konflux Community Deployment via ArgoCD 
-```console
+```bash
 ./deploy-argocd.sh <ArgoCD server> <production/staging>
 ```
 
 ## Accessing the Konflux Console
-```console
+```bash
 kubectl port-forward -n konflux-ui svc/proxy 9443:9443 
-```
\ No newline at end of file
+```
+
+## Setting up a development environment using kind
+
+### Installing Software Dependencies
+Verify that the applications below are installed on the host machine:
+
+* [Kind and kubectl](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+  along with `podman` or `docker`
+
+### Bootstrapping the development cluster
+Clone this repository:
+
+ ```bash
+git clone https://github.com/konflux-ci/infra-deployments
+cd infra-deployments
+```
+
+**Note:** It is recommended that you increase the `inotify` resource limits in order to
+avoid issues related to
+[too many open files](https://kind.sigs.k8s.io/docs/user/known-issues/#pod-errors-due-to-too-many-open-files). To increase the limits temporarily, run the
+following commands:
+
+```bash
+sudo sysctl fs.inotify.max_user_watches=524288
+sudo sysctl fs.inotify.max_user_instances=512
+```
+
+From the root of this repository, run the setup scripts:
+
+1. Create a cluster
+
+```bash
+kind create cluster --name dev --config kind-config.yaml
+```
+
+**Note:** When using Podman, it is recommended that you increase the PID limit on the
+container running the cluster, as the default might not be enough when the cluster
+becomes busy:
+
+```bash
+podman update --pids-limit 4096 dev-control-plane
+```
+
+**Note:** If pods still fail to start due to missing resources, you may need to reserve
+additional resources to the Kind cluster. Edit [kind-config.yaml](./kind-config.yaml)
+and modify the `system-reserved` line under `kubeletExtraArgs`:
+
+```yaml
+  kubeadmConfigPatches:
+  - |
+    kind: InitConfiguration
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: "ingress-ready=true"
+        system-reserved: memory=12Gi
+```
+
+2. Create argocd namespace in the cluster:
+```bash
+kubectl create namespace argocd
+```
+
+3. Deploy ArgoCD:
+```bash
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+```
+
+4. Once ArgoCD is up and running, obtain its server's password:
+```bash
+kubectl -n argocd get secret argocd-initial-admin-secret -o jsonpath="{.data.password}" | base64 -d && echo
+```
+This password will be used to login to ArgoCD UI as well as for deploying konflux.
+
+5. To deploy konflux, we need to have access to ArgoCD server, we will port forward ArgoCD server:
+```bash
+kubectl port-forward -n argocd service/argocd-server 8443:443
+```
+
+6. Update the `login_argocd` method in the `deploy-argocd.sh` script as follows:
+```bash
+login_argocd() {
+    ARGOCD_SERVER=$(echo "$1"|tr '[:upper:]' '[:lower:]')
+    argocd login $ARGOCD_SERVER --username admin --password <ArgoCD server password> --insecure
```

Comment:
I'm not sure this is ideal. People may accidentally commit their passwords to git and even include it in a PR. I know this password is not very critical, but I think we should try to avoid this.

Reply:
I updated the instructions
How is it now?

Reply:
I like the approach of manual login. See my suggestion about changing the original line in the script, so that the user won't have to edit the script at all.